doc/UsersGuide/source/BuildingRunningTesting/HAFSCodeContributionGuidelines.rst

```diff
@@ -0,0 +1,82 @@
+- Users submit a pull request (PR) for code commit.
+- Pull request creators will add reviewers, if needed, ask code managers to add reviewers.
+- At least one reviewer with write permission needs to give approval before the code can be committed.
+- If needed, developers can ask code manager(s) to run the regression test on the required platforms that users don’t have access to.
+- When the PR is reviewed and no further changes are required, the PR can be put in the commit queue.
+- For PRs that involve several repositories:
+  i) All PRs need to be cross-referenced in the description of each repository PR; and a note needs to be placed in each PR.
+  ii) During the code commit, all PRs need to be ready for commit. This requires that all the related commit branches are merged with the top of repositories and no other commit will be committed into each repository until the current commit is finished (repository freeze for the commit). If any of the repositories is updated, the code manager of that repository needs to notify all the related repository code managers to redo RT.
+  iii) The top application RTs need to be finished and RT information needs to be posted in all the related repositories before the PR can be merged in each individual repository.
+
+How to Write Pull Request
+--------------------------
+
+- Pull request size: should be small
+- Feature breaking: if possible, one feature for each PR..
+- Make a self-explanatory title describing what the PR does.
+- Description: Details with what was changed, why it was changed, and how it was changed
+
+Code Commit
+-----------
+
+- The code changes will be reviewed and approved by at least one code reviewer.
+- The CI tests need to pass.
+- Regression tests need to pass on all the supported platforms. All the regression test log files need to be updated. The exception is when a certain platform is under maintenance. For such a time, the regression test will be skipped on that platform. When the platform becomes available, the following PR will be tested and verified. If issues come up with the platform, then PRs will be suspended until the issue is resolved.
+- At least one code manager needs to review and approve the code changes before the code is merged to the develop branch.
+- Code managers will discuss the current PRs in UFS and sub-component repositories and a commit queue will be determined and put in the ufs wiki page.
+
+Commit Procedure
+----------------
+
+- For any PR created in the app or subcomponent repo (assume proper testing is done for the code updates), code managers needs to check:
+  - The PR template instruction is followed, required information is provided.
+  - The code is merged to the top of develop/master branch
+  - Put labels such as “documentation”, “bug”, “enhancement”, “baseline change”, “no baseline change”, “New Input Data Req’d”, “Input data change”
+  - Put “Tri
```

Comment:
@jdkublnick This seems to have gotten truncated? Also, I don't think the HAFS repo has a documentation label yet. You'll need to check that all these labels actually exist in HAFS, since each repo has slightly different labels.